aiven/client/client.py

```diff
@@ -174,11 +179,28 @@ def verify(
                 )
                 time.sleep(0.2)
 
+        return self._process_response(response=response, op=op, path=path, result_key=result_key)
+
+    @staticmethod
+    def build_path(*parts: str) -> str:
+        return "/" + "/".join(quote(part, safe="") for part in parts)
+
+    def _process_response(
+        self,
+        response: Response,
+        op: Callable[..., Response],
+        path: str,
+        result_key: str | None = None,
+    ) -> Mapping | str:
         # Check API is actually returning data or not
         if response.status_code == HTTPStatus.NO_CONTENT or response.headers.get("Content-Length", "0") == "0":
             return {}
 
-        result = response.json()
+        try:
+            result = response.json()
```

Comment:
I think a choice between decoding and not decoding this should be done based on response `Content-Type` so that 1. we don't decode and thus modify something that just happens to look like JSON if it's not intended to be processed as such, and 2. if backend malfunctions and returns broken or incomplete JSON, we can fail instead of passing broken data to e.g. another process in a shell pipeline that may not be prepared for such.

Especially if there's custom file data as `application/octet-stream`, I wouldn't expect client to decode it and output a possibly modified version of the contents, even if it happens to look like JSON.

Reply:
Makes sense, changed it, thank you